subsys/zbus/Kconfig

```diff
@@ -9,6 +9,10 @@ menuconfig ZBUS
 
 if ZBUS
 
+config ZBUS_PREFER_DYNAMIC_ALLOCATION
+	bool "Set zbus to work on static allocation only"
+	default y if HEAP_MEM_POOL_SIZE > 0
```

Comment:
The system heap can still exist when `HEAP_MEM_POOL_SIZE == 0`, the size of the heap is the sum of `HEAP_MEM_POOL_SIZE` and all the `HEAP_MEM_POOL_ADD_SIZE_` options.

Reply:
I see. Do you have any suggestions on how to do that correctly?

Reply:
I set the configuration to "default y". If the user does not want to use the heap, it is possible to set it accordingly. 
```kconfig
config ZBUS_PREFER_DYNAMIC_ALLOCATION
	bool "Set zbus to work with dynamic allocation using the system heap"
	default y
```